egg/core/trainers.py

```diff
@@ -168,7 +169,9 @@ def eval(self):
         self.game.eval()
         with torch.no_grad():
             for batch in self.validation_data:
-                batch = move_to(batch, self.device)
+                if not isinstance(batch, Batch):
+                    batch = Batch(*batch)
+                batch.to(self.device)
```

Comment:
I think here you create a copy so you need to assign 
```suggestion
                batch = batch.to(self.device)
```